src/CTA.Rules.Config/Utils.cs

```diff
@@ -191,20 +192,52 @@ public static string ThreadSafeExportStringToFile(string filePath, string conten
             }
         }
 
+
+        public static string CopySolutionToDestinationLocation(string solutionName, string solutionPath)
+        {
+            string slnDirPath = Directory.GetParent(solutionPath).FullName;
+            string root = Path.GetPathRoot(slnDirPath);
+            string relativeSrc = Path.GetRelativePath(root, slnDirPath);
+            
+            var newTempDirPath = Path.Combine(Directory.GetParent(slnDirPath).FullName, Guid.NewGuid().ToString());
+            string destPath = Path.Combine(newTempDirPath, relativeSrc);
+            string newSlnPath = CopyFolderToTemp(solutionName, slnDirPath, destPath);
+
+
+            if (solutionPath.Contains(".sln") && File.Exists(solutionPath))
+            {
+                IEnumerable<string> projects = GetProjectPathsFromSolutionFile(solutionPath);
+                foreach (string project in projects)
+                {
+                    string projPath = Directory.GetParent(project).FullName;
+                    relativeSrc = Path.GetRelativePath(solutionPath, projPath);
+                    string projName = Path.GetFileName(project);
```

Comment:
I would move this line and the line above it into the if block, since they aren't needed if the condition isn't true.